Allow choice of qemu binary and acceleration

rn opened the pull request:
There are more acceleration backend in the pipeline for qemu. Allow to select the qemu binary being used (makes it easier to select master builds) and the acceleration option.

For example:
```
linuxkit run qemu -qemu ~/src/repos/virt/qemu/x86_64-softmmu/qemu-system-x86_64 -accel hvf minimal
```
boots with a qemu master build on macOS with Hypervisor Framework Acceleration.

Note this also changes the `LINUXKIT_QEMU_KVM` env variable to `LINUXKIT_QEMU_ACCEL`

![turtle](https://user-images.githubusercontent.com/3338098/35690971-30d12118-076f-11e8-80f9-ec3c5ed61a74.jpg)

ijc commented on src/cmd/linuxkit/run_qemu.go:
```diff
@@ -160,13 +151,14 @@ func runQemu(args []string) {
 
 	// Backend configuration
 	qemuContainerized := flags.Bool("containerized", false, "Run qemu in a container")
+	qemuCmd := flags.String("qemu", "", "Path to the qemu binary (if not in default location)")
```
"if not in $PATH" might be a more accurate way of putting it?

ijc commented on src/cmd/linuxkit/run_qemu.go:
```diff
@@ -499,12 +492,12 @@ func buildQemuCmdline(config QemuConfig) (QemuConfig, []string) {
 			qemuArgs = append(qemuArgs, "-machine", "virt,gic_version=host")
 		} else {
-			qemuArgs = append(qemuArgs, "-machine", "q35,accel=kvm:tcg")
+			qemuArgs = append(qemuArgs, "-machine", "q35")
```
IIRC `accel=kvm:tcg` meant "try kvm but fallback to tcg" or some such, so should the default for `config.Accel` be `kvm:tcg` rather than just `kvm`?

rn commented:
turns out `-accel` is relatively new, for example not in 2.5.0 which is shipping with Ubuntu 16.04. So need to rework this to use `-machine blah,accel=`

rn commented:
Pushed update:
- fixed the nits from @ijc 
- use the `accel` option for system

Also enabled `hvf` as default on macOS and fixed a bug where kvm was selected for non-native architectures, ie running x86 on arm64 and vice versa

Tested on macOS, Linux x86 and arm64 with both x86 and arm64 images

So this is ready to be reviewed
